addons/reloadlaunchers/functions/fnc_canLoad.sqf

```diff
@@ -4,38 +4,38 @@
  * Check of the unit can reload the launcher of target unit.
  *
  * Arguments:
- * 0: Unit to do the reloading <OBJECT>
- * 1: Unit eqipped with launcher <OBJECT>
- * 2: weapon name <STRING>
- * 3: missile name <STRING>
+ * 0: Unit to execute the reload <OBJECT>
+ * 1: Unit equipped with the launcher <OBJECT>
+ * 2: Launcher name <STRING>
+ * 3: Missile name <STRING>
  *
  * Return Value:
  * None
  *
  * Example:
- * [bob, kevin, "weapon", "missile"] call ace_reloadlaunchers_fnc_canLoad
+ * [player, cursorTarget, "launch_RPG32_F", "RPG32_F"] call ace_reloadlaunchers_fnc_canLoad
  *
  * Public: No
  */
 
 params ["_unit", "_target", "_weapon", "_magazine"];
 TRACE_4("params",_unit,_target,_weapon,_magazine);
 
-if (!alive _target) exitWith {false};
-if (vehicle _target != _target) exitWith {false};
+// Target must be awake
+if !(_target call EFUNC(common,isAwake)) exitWith {false};
+
+// Target must not be in a vehicle
+if !(isNull objectParent _target) exitWith {false};
 if !([_unit, _target, ["isNotInside", "isNotSwimming"]] call EFUNC(common,canInteractWith)) exitWith {false};
 
-// target is awake
-if (_target getVariable ["ACE_isUnconscious", false]) exitWith {false};
+// Check if the launcher is compatible
+if (getNumber (configFile >> "CfgWeapons" >> _weapon >> QGVAR(enabled)) == 0) exitWith {false};
 
-// has secondary weapon equipped
+// Check if target has its secondary weapon equipped
 if !(_weapon in weapons _target) exitWith {false};
 
-// check if the target really needs to be reloaded
-if (count secondaryWeaponMagazine _target > 0) exitWith {false};
-
-// check if the launcher is compatible
-if (getNumber (configFile >> "CfgWeapons" >> _weapon >> QGVAR(enabled)) == 0) exitWith {false};
+// Check if the target's launcher's primary muzzle really needs to be reloaded
+if (((getUnitLoadout _target) select 1 param [4, []] param [1, 0]) != 0) exitWith {false};
```

Comment:
Oof. `getUnitLoadout` is *slow* for a condition. Maybe use `weaponsInfo` with 2.14 changes (or `weaponState`) instead?

Reply:
I might be able to get away with `_target ammo _weapon`, since we are checking primary muzzles only, but I need to check more.

Reply:
> I might be able to get away with `_target ammo _weapon`, since we are checking primary muzzles only, but I need to check.

Testing suggests `_target ammo _weapon` works.

Reply:
I was digging into `ace_reload`, as I'm cleaning up its code too. I stumbled across https://github.com/acemod/ACE3/issues/1119. I can confirm that the issue described in https://github.com/acemod/ACE3/issues/1119 still exists. It also happens when `player setUnitLoadout getUnitLoadout player` is used.
`weaponState` in those cases also returns incorrect information.

A ticket can be found here: https://feedback.bistudio.com/T167015, to which I'll add my findings.

But because of this, I'm not sure if using `_target ammo _weapon` is safe.